packages/drafts-realm/campaign-form.gts

```diff
@@ -0,0 +1,518 @@
+    return '';
+  }
+
+  let currentNumber = num;
+  if (typeof num === 'string') {
+    currentNumber = parseFloat(num);
+  }
+
+  return `${currentNumber.toLocaleString('en-US')}${isPercentage ? ' %' : ''}`;
+};
+
+class Isolated extends Component<typeof CampaignForm> {
+  <template>
+    <div class='campaign-form-isolated'>
+      <FieldContainer @label='Name' class='field'>
+        {{@model.name}}
+      </FieldContainer>
+      <FieldContainer @label='Status' class='field'>
+        {{@model.status}}
+      </FieldContainer>
+      <FieldContainer @label='Active' class='field'>
+        {{if @model.active 'Yes' 'No'}}
+      </FieldContainer>
+      <FieldContainer @label='Type' class='field'>
+        {{@model.type}}
+      </FieldContainer>
+      <FieldContainer @label='Parent Campaign' class='field'>
+        <@fields.parent_campaign />
+      </FieldContainer>
+      <FieldContainer @label='Description' class='field'>
+        {{@model.description}}
+      </FieldContainer>
+      <FieldContainer @label='Start Date' class='field'>
+        <@fields.start_date />
+      </FieldContainer>
+      <FieldContainer @label='End Date' class='field'>
+        <@fields.end_date />
+      </FieldContainer>
+      <FieldContainer @label='Num Sent in Campaign' class='field'>
+        {{formatNumberWithSeparator @model.number_sent}}
+      </FieldContainer>
+      <FieldContainer @label='Expected Response (%)' class='field'>
+        {{formatNumberWithSeparator @model.expected_response_percentage true}}
+      </FieldContainer>
+      <FieldContainer @label='Expected Revenue in Campaign' class='field'>
+        {{formatCurrency @model.expected_revenue}}
+      </FieldContainer>
+      <FieldContainer @label='Budgeted Cost in Campaign' class='field'>
+        {{formatCurrency @model.budgeted_cost}}
+      </FieldContainer>
+      <FieldContainer @label='Actual Cost in Campaign' class='field'>
+        {{formatCurrency @model.actual_cost}}
+      </FieldContainer>
+    </div>
+    <style>
+      .campaign-form-isolated {
+        display: grid;
+        gap: var(--boxel-sp-lg);
+        padding: var(--boxel-sp-xl);
+      }
+    </style>
+  </template>
+}
+
+class Embedded extends Component<typeof CampaignForm> {
+  <template>
+    {{@model.name}}
+  </template>
+}
+
+class Edit extends Component<typeof CampaignForm> {
+  @tracked name = this.args.model.name;
+  @tracked selectedStatus = { name: this.args.model.status };
+  @tracked selectedActive = this.args.model.active;
+  @tracked selectedType = { name: this.args.model.type };
+  @tracked description = this.args.model.description;
+  @tracked startDateString = this.args.model.start_date
+    ? format(this.args.model.start_date, dateFormat)
+    : null;
+  @tracked endDateString = this.args.model.end_date
+    ? format(this.args.model.end_date, dateFormat)
+    : null;
+  @tracked numberSentInputValue = formatNumberWithSeparator(
+    this.args.model.number_sent,
+  );
```

Comment:
Why is there 2 places formatting. At a tracked property definition and the isolated view. The final formatted value is good to be specified inside of a `get()` since it is derived from this.args.model.number_sent.

Reply:
If possible can we reduce the usage of @tracked states, `this.args.model` I believe should reactive. We derive that data.